File: meme_inator_front/lib/features/profiles/domain/usecases/patch_my_profile_usecase.dart
import 'package:meme_inator_front/core/iusecase.dart';
import 'package:meme_inator_front/core/results.dart';
import 'package:meme_inator_front/features/profiles/domain/entities/profile_entity.dart';
import 'package:meme_inator_front/features/profiles/domain/interfaces/iprofile_repository.dart';
import 'package:uuid/uuid.dart';

class PatchMyProfileInput {
  final Uuid userId;
  final Map<String, dynamic> partialData;

  const PatchMyProfileInput({
    required this.userId,
    required this.partialData,
  });
}

class PatchMyProfileUsecase
    extends IUseCase<PatchMyProfileInput, ProfileEntity> {
  final IProfileRepository profileRepo;

  PatchMyProfileUsecase(this.profileRepo);

  @override
  Future<Result<ProfileEntity>> execute({
    required PatchMyProfileInput valObj,
  }) async {
    throw UnimplementedError('PatchMyProfileUsecase.execute not implemented');
  }
}

File: meme_inator_back/apps/profiles/application/usecases/patch_my_profile_usecase.py
from typing import Dict, Any
from uuid import UUID
from apps.profiles.domain.entities.profile_entity import ProfileEntity
from apps.profiles.domain.usecases.ipatch_my_profile_usecase import IPatchMyProfileUsecase
from apps.profiles.domain.irepositories.i_profile_repository import IProfileRepository
from core.results import Result

class PatchMyProfileUsecase(IPatchMyProfileUsecase):
    def __init__(self, profile_repo: IProfileRepository):
        self.profile_repo = profile_repo

    def execute(self, user_id: UUID, partial_data: Dict[str, Any]) -> Result[ProfileEntity]:
        # Repository expects the partial data (already in camelCase as sent from frontend)
        return self.profile_repo.patch_my_profile(user_id, partial_data)